pkg/messaging/nats/pubsub.go

```diff
@@ -162,6 +162,9 @@ func (ps *pubsub) natsHandler(h messaging.MessageHandler) broker.MsgHandler {
 			ps.logger.Warn(fmt.Sprintf("Failed to unmarshal received message: %s", err))
 			return
 		}
+		if msg.Subtopic == "" {
+			msg.Subtopic = m.Subject // set for messages not published using adapters.
```

Comment:
Why is this needed? This is confusing.

If there is no subtopic, it should stay an empty string.